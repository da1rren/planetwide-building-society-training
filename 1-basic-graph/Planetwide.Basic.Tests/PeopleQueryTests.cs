using System.Text.Json;
using System.Text.Json.Serialization;
using HotChocolate.Execution;
using Microsoft.AspNetCore.Mvc.Testing;

namespace Planetwide.Basic.Tests;

public class PeopleQueryTests
{
    public PeopleQueryTests()
    {

    }

    [Fact]
    public async Task Ensure_Get_Person_Returns_Person()
    {
        // The web app factory should be extracted to a fixture 
        var application = new WebApplicationFactory<Program>()
            .WithWebHostBuilder(builder =>
            {
                // ... Configure test services
            });

        var client = application.CreateClient();

        var request = await client.PostAsJsonAsync("/graphql", new
        {
            query = @"
                { person { name }}
            "
        });

        var response = await request.Content.ReadAsStringAsync();
        request.EnsureSuccessStatusCode();
        Snapshot.Match(response);
    }
}

