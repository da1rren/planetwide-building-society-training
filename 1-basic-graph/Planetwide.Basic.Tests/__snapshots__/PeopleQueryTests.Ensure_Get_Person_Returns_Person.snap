{"data":{"person":{"name":"Luke Skywalker"}}}
